imap_processing/ultra/l2/ultra_l2.py

```diff
@@ -0,0 +1,334 @@
+        # (in that order to avoid double weighting by solid angle)
+        for (
+            quantity_to_weight
+        ) in VARIABLES_TO_WEIGHT_BY_POINTING_SET_EXPOSURE_TIMES_SOLID_ANGLE:
+            pointing_set.data[quantity_to_weight] = (
+                pointing_set.data[quantity_to_weight]
+                * pointing_set.data["pointing_set_exposure_times_solid_angle"]
+            )
+
+        skymap.project_pset_values_to_map(
+            pointing_set=pointing_set,
+            value_keys=list(
+                set(
+                    output_map_properties.values_to_push_project
+                    + REQUIRED_L1C_VARIABLES
+                )
+            ),
+            index_match_method=ena_maps.IndexMatchMethod.PUSH,
+        )
+
+    # Subsequent processing for weighted quantities at SkyMap level
+    for (
+        quantity_to_weight
+    ) in VARIABLES_TO_WEIGHT_BY_POINTING_SET_EXPOSURE_TIMES_SOLID_ANGLE:
+        skymap.data_1d[quantity_to_weight] = (
+            skymap.data_1d[quantity_to_weight]
+            / skymap.data_1d["pointing_set_exposure_times_solid_angle"]
+        )
+
+    # TODO: Ask Ultra team about this - I think this is a decent
+    # but imperfect approximation for meaning the exposure:
+    # (dividing by the 1/(number of PSETs)) to fix the exposure being mean-ed over
+    # all pixels in all PSETs which feed into a map superpixel,
+    # rather than being mean-ed over pixels in a PSET and summed over PSETs
+    skymap.data_1d["exposure_factor"] /= skymap.data_1d[
+        "num_pointing_set_pixel_members"
+    ] / len(ultra_l1c_psets)
+
+    # TODO: Ask Ultra team about background rates - I think they should increase when
+    # binned to larger pixels, as I've done here, but that was never explicitly stated
+    skymap.data_1d["background_rates"] *= skymap.solid_angle / pointing_set.solid_angle
+
+    # Get the energy bin widths from a PointingSet (they will all be the same)
+    delta_energy = pointing_set.data["energy_bin_delta"]
+
+    # Core calculations of flux and flux uncertainty for L2
+    # Get corrected count rate with background subtraction applied
+    skymap.data_1d["corrected_count_rate"] = (
+        skymap.data_1d["counts"] / skymap.data_1d["exposure_factor"]
+    ) - skymap.data_1d["background_rates"]
+
+    # Calculate flux as corrected_counts / (sensitivity * solid_angle * delta_energy)
+    skymap.data_1d["flux"] = skymap.data_1d["corrected_count_rate"] / (
+        skymap.data_1d["sensitivity"] * skymap.solid_angle * delta_energy
+    )
+
+    skymap.data_1d["flux_uncertainty"] = (skymap.data_1d["counts"] ** 0.5) / (
+        skymap.data_1d["exposure_factor"]
+        * skymap.data_1d["sensitivity"]
+        * skymap.solid_angle
+        * delta_energy
+    )
+
+    # Drop the variables that are no longer needed
+    skymap.data_1d = skymap.data_1d.drop_vars(
+        VARIABLES_TO_DROP_AFTER_FLUX_CALCULATION,
+        errors="ignore",
```

Comment:
What errors are being ignored? Do we need to log anything here so we aren't missing something, or add a comment indicating why this is used.

Reply:
My bad - this is not necessary anymore after a change I made!